ofrak_core/ofrak/core/gzip.py

```diff
@@ -41,44 +53,27 @@ class GzipUnpacker(Unpacker[None]):
 
     async def unpack(self, resource: Resource, config=None):
         data = await resource.get_data()
-        # GzipFile is faster (spawning external processes has overhead),
-        # but pigz is more willing to tolerate things like extra junk at the end
-        try:
-            with GzipFile(fileobj=BytesIO(data), mode="r") as gzip_file:
-                return await resource.create_child(
-                    tags=(GenericBinary,),
-                    data=gzip_file.read(),
-                )
-        except BadGzipFile:
-            # Create temporary file with .gz extension
-            with tempfile.NamedTemporaryFile(suffix=".gz") as temp_file:
-                temp_file.write(data)
-                temp_file.flush()
-                cmd = [
-                    "pigz",
-                    "-d",
-                    "-c",
-                    temp_file.name,
-                ]
-                proc = await asyncio.create_subprocess_exec(
-                    *cmd,
-                    stdout=asyncio.subprocess.PIPE,
-                    stderr=asyncio.subprocess.PIPE,
-                )
-                stdout, stderr = await proc.communicate()
-                data = stdout
-                if proc.returncode:
-                    # Forward any gzip warning message and continue
-                    if proc.returncode == -2 or proc.returncode == 2:
-                        LOGGER.warning(stderr)
-                        data = stdout
-                    else:
-                        raise CalledProcessError(returncode=proc.returncode, cmd=cmd, stderr=stderr)
-
-                await resource.create_child(
-                    tags=(GenericBinary,),
-                    data=data,
-                )
+        unpacked_data = await self.unpack_with_zlib_module(data)
+        return await resource.create_child(tags=(GenericBinary,), data=unpacked_data)
+
+    @staticmethod
+    async def unpack_with_zlib_module(data: bytes) -> bytes:
+        chunks = []
+
+        # wbits > 16 handles the gzip header and footer
+        # We need to create a zlib.Decompress object in order to use this
+        # parameter in Python < 3.11
+        decompressor = zlib.decompressobj(wbits=16 + zlib.MAX_WBITS)
+        while data.startswith(b"\037\213"):
+            chunks.append(decompressor.decompress(data))
+            if decompressor.eof:
+                break
+            data = decompressor.unused_data.lstrip(b"\0")
```

Comment:
I think this comment is very unclear as it stands. My understanding is that it means "this is why we're using `zlib.decompressobj`," but it reads like "this might not work in Python 3.11 because we aren't using a `zlib.Decompress` object."

I would also amend the comment to cite a source (or include justification) for why we're doing decompression this way here. To a future reader, I don't think it will be obvious why we're decompressing data that begins with these magic bytes in chunks like this.